pkg/controller/eks/fargateprofile/controller.go

```diff
@@ -0,0 +1,191 @@
+	name := managed.ControllerName(v1alpha1.FargateProfileKind)
+
+	return ctrl.NewControllerManagedBy(mgr).
+		Named(name).
+		For(&v1alpha1.FargateProfile{}).
+		Complete(managed.NewReconciler(mgr,
+			resource.ManagedKind(v1alpha1.FargateProfileGroupVersionKind),
+			managed.WithExternalConnecter(&connector{kube: mgr.GetClient(), newEKSClientFn: eks.NewEKSClient}),
+			managed.WithInitializers(managed.NewDefaultProviderConfig(mgr.GetClient()), managed.NewNameAsExternalName(mgr.GetClient()), &tagger{kube: mgr.GetClient()}),
+			managed.WithReferenceResolver(managed.NewAPISimpleReferenceResolver(mgr.GetClient())),
+			managed.WithLogger(l.WithValues("controller", name)),
+			managed.WithRecorder(event.NewAPIRecorder(mgr.GetEventRecorderFor(name)))))
+}
+
+type connector struct {
+	kube           client.Client
+	newEKSClientFn func(config aws.Config) eks.Client
+}
+
+func (c *connector) Connect(ctx context.Context, mg resource.Managed) (managed.ExternalClient, error) {
+	cr, ok := mg.(*v1alpha1.FargateProfile)
+	if !ok {
+		return nil, errors.New(errNotEKSFargateProfile)
+	}
+	cfg, err := awsclients.GetConfig(ctx, c.kube, mg, cr.Spec.ForProvider.Region)
+	if err != nil {
+		return nil, err
+	}
+	return &external{client: c.newEKSClientFn(*cfg), kube: c.kube}, nil
+}
+
+type external struct {
+	client eks.Client
+	kube   client.Client
+}
+
+func (e *external) Observe(ctx context.Context, mg resource.Managed) (managed.ExternalObservation, error) {
+	cr, ok := mg.(*v1alpha1.FargateProfile)
+	if !ok {
+		return managed.ExternalObservation{}, errors.New(errNotEKSFargateProfile)
+	}
+
+	rsp, err := e.client.DescribeFargateProfileRequest(&awseks.DescribeFargateProfileInput{FargateProfileName: aws.String(meta.GetExternalName(cr)), ClusterName: &cr.Spec.ForProvider.ClusterName}).Send(ctx)
+	if err != nil {
+		return managed.ExternalObservation{}, errors.Wrap(resource.Ignore(eks.IsErrorNotFound, err), errDescribeFailed)
+	}
+
+	current := cr.Spec.ForProvider.DeepCopy()
+	eks.LateInitializeFargateProfile(&cr.Spec.ForProvider, rsp.FargateProfile)
+
+	cr.Status.AtProvider = eks.GenerateFargateProfileObservation(rsp.FargateProfile)
+	// Any of the statuses we don't explicitly address should be considered as
+	// the fargate profile being unavailable.
+	switch cr.Status.AtProvider.Status { // nolint:exhaustive
+	case v1alpha1.FargateProfileStatusActive:
+		cr.Status.SetConditions(runtimev1alpha1.Available())
+	case v1alpha1.FargateProfileStatusCreating:
+		cr.Status.SetConditions(runtimev1alpha1.Creating())
+	case v1alpha1.FargateProfileStatusDeleting:
+		cr.Status.SetConditions(runtimev1alpha1.Deleting())
+	default:
+		cr.Status.SetConditions(runtimev1alpha1.Unavailable())
+	}
+
+	return managed.ExternalObservation{
+		ResourceExists:          true,
+		ResourceUpToDate:        eks.IsFargateProfileUpToDate(cr.Spec.ForProvider, rsp.FargateProfile),
+		ResourceLateInitialized: !reflect.DeepEqual(current, &cr.Spec.ForProvider),
```

Comment:
How about using `cmp.Equal` here for consistency?

Reply:
done